### DISCUSSION TITLE
Feature Request (Github Actions): Dynamic Workflow Run Name

### DISCUSSION THREAD

**Discussion by mpmarven-ms** [community/community#11396] [Actions] [answered]
It would be nice to be able to add meaningful runtime data to the name ie via something similar to how a job can be named dynamically.  

Something like: name: "Build and Deploy to ${{github.event.inputs.environment}}".

**Comment by ethomson** [comment]
Thanks for the feedback @mpmarven-ms - I agree with you that it would be nice to be able to configure display names.  We'll take a look at adding this support.

**Reply by theopoenaru** [reply]
hi @ethomson, would it be possible to get an update if this is being prioritized? I have the same issue as @philomory below: when triggering a workflow based on an event, the name will be a smart default (e.g. the triggering commit), however when triggering manually, it simply names it the workflow name. Not even the branch is shown for those instances, even though you can still filter manually triggered runs by branch using the search bar. Because of these issues I cannot use the actions list, and have to repeatedly click into individual runs when searching for a specific one. 

I would like to include an input variable in the title so that I can scan the actions list. I would also like to see the branch it was run off of, similar to how event triggered runs show up.

e.g. see how the first run (event triggered) is usefully auto-labelled but the next five (manually triggered) are indistinguishable.
<img width="922" alt="Screen Shot 2022-08-09 at 11 11 37 PM" src="https://user-images.githubusercontent.com/9571802/183802154-1bc40505-cbe8-4ce0-a12f-b07ea15ca13a.png">

**Comment by philomory** [comment]
This is especially crucial for workflow runs triggered by `workflow_dispatch` events, since currently in such workflows every run of a given workflow is completely indistinguishable from one another in the "runs" list.

**Comment by AvivPeledTalon** [comment]
This would really help my team.
Our current solution is an in-house browser extension parsing the info we wish to display, and editing Github's HTML to show it while browsing the "Actions" section

**Reply by sushiljainam-gt** [reply]
can this browser extension be used by other people as well? can it be open sourced? @AvivPeledTalon

**Comment by KevinMSampson** [comment]
I would suggest dynamic folders instead.
```
name: Primary Workflow
sub-name: ${{ inputs.environment }}
```
When looking at your Workflows in the Actions tab:
- Primary Workflow
  - PROD
  - UAT
  - DEV
- Secondary Workflow

**Reply by FrancisVillarba** [reply]
Why not both?

**Comment by dong-pham-ts** [comment]
LGTM .My team need to review what action have been done so we should add dynamic Workflow Run Name to review that

**Comment by mike-potter** [comment]
> This is especially crucial for workflow runs triggered by workflow_dispatch events, since currently in such workflows every run of a given workflow is completely indistinguishable from one another in the "runs" list.

This is also the issue we have.  Drives the developers crazy not being able to distinguish the runs in the Actions list.  I can understand restrictions on the variables allowed here, but seems like the inputs from a triggered workflow_dispatch should be available when the workflow run is named.

If the workflow name can't be dynamic, some way to add to the description to show the inputs would be helpful.

**Comment by mbrancato** [comment]
Possibly a different suggestion... add an optional `title` body parameter. Then just update libraries to support this.

https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event

```
curl \
  -X POST \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/actions/workflows/WORKFLOW_ID/dispatches \
  -d '{"title":"My custom workflow title","ref":"topic-branch","inputs":{"name":"Mona the Octocat","home":"San Francisco, CA"}}'
```

**Reply by wafadje732-byte** [reply]
curl \
  -X POST \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/actions/workflows/WORKFLOW_ID/dispatches \
  -d '{"title":"My custom workflow title","ref":"topic-branch","inputs":{"name":"Mona the Octocat","home":"San Francisco, CA"}}'

**Comment by mashhurs** [comment]
That would be a great feature.
Whenever I take a look at workflow runs, they all look same except small ID run difference which doesn't make any sense for our use case. I would highly appreciate if we can attach user defined texts at run time to the workflow name or description.
I mean by description, the red text in the pic:
<img width="563" alt="Screen Shot 2022-08-31 at 2 47 39 PM" src="https://user-images.githubusercontent.com/99575341/187791188-e1790b41-6365-405a-99d7-3db29c162127.png">

**Comment by andrewakim** [comment, marked as answer]
Hi All - Just wanted to let everyone know that [this feature is now live](https://github.blog/changelog/2022-09-26-github-actions-dynamic-names-for-workflow-runs/).

**Reply by strangesource** [reply]
Great to see this ❤️ ! Thanks to everyone involved.

One follow up question which I was not able to answer from the documentation:
Is it possible to set the `run-name` to a value from a job output?

**Reply by andrewakim** [reply]
Hi @strangesource - The new `run-name` feature does not support the `needs` context.  It can get a bit tricky if we allow `run-name` to be set/updated during a workflow run but I'm still thinking it through.

**Reply by chehsunliu** [reply]
Wish this feature could be enabled on GHES!

**Reply by Drapegnik** [reply]
@andrewakim, 
Thanks for the feature!

Are there any plans to make `env` context available in the `run-name`?

**Reply by sergey-koryshev** [reply]
Agreed, it would be great to have ability to set `run-name` during step execution. For example: when you read version from file and then pass it to `run-name` via `env` context or by another way.

**Reply by rokeno** [reply]
> Wish this feature could be enabled on GHES!

Why is this not applied to Enterprise as well?
What am i missing here?

GHES is Github as much as Github public.

**Reply by andrewakim** [reply]
Hi @rokeno & @chehsunliu - This feature will be included in the next version of GHES (3.8).  Thanks!

**Reply by MrHinsh** [reply]
I want to update the run name from within a job. I use GitVersion to get the version and want the run-name to reflect the semantic version umber being built

**Comment by umassahu** [comment]
Thank you to each and everyone for making this happen , superb feature and helping in multiple ways

**Comment by sammcj** [comment]
I noticed this was missing from schemastore so I've submitted a PR to update the Github Workflow JSON schema: https://github.com/SchemaStore/schemastore/pull/2503

**Comment by octavianandrei** [comment]
Is there a way to filter workflows by name using run-name feature? as far as I can see its not possible since it will still get the workflow name by default.
It was nice to have this feature since for example I can have 100 runs on an workflow triggered from other repos and I want to sort a specific values in "filter workflow runs"

**Comment by cgunning** [comment]
This is a great feature but we use reusable workflows a lot and from my tests it does not seem to be possible to set the `run-name` from within the reusable workflow. Are there any plans to support this?

**Reply by cgunning** [reply]
After thinking a bit more about this I realize that since a workflow can use multiple reusable workflows in different jobs it doesn't really make sense even though it would have been really nice for our use case where we only use one reusable workflow from a workflow.

**Comment by dvag-dominik-schumacher** [comment]
Thanks for implementing this feature! It's important for our developers experience! :-)

**Comment by schr4der** [comment]
I'm having issues using this on an enterprise repo. Are there plans to implement this for enterprise?

**Comment by rokeno** [comment]
They will be in version 3.8
Check your company's GHES update rollout.

**Comment by krishanuc1001** [comment]
Hey there folks!!! Is there a way to use the job names as the run-names dynamically in a workflow file having multiple jobs?

**Comment by Lewluu** [comment]
is there any way to get the run-name from POST response data?
e.g.: "workflow_run['run-name']"
workflow_run['display_title'] gives commit message
